app/src/main/java/com/babylon/wallet/android/presentation/settings/linkedconnectors/qrcode/CameraPreview.kt

```diff
@@ -55,13 +56,16 @@ private fun BarcodePreviewView(
                 }
                 val cameraProvider: ProcessCameraProvider = cameraProviderFuture.get()
 
-                val barcodeAnalyser = QrcodeAnalyser { qrCodes ->
-                    qrCodes.forEach { barcode ->
-                        barcode.rawValue?.let { barcodeValue ->
-                            onQrCodeDetected(barcodeValue.replace("radix:", ""))
+                val barcodeAnalyser = QrcodeAnalyser(
+                    onBarcodeDetected = { qrCodes ->
+                        qrCodes.forEach { barcode ->
+                            barcode.rawValue?.let { barcodeValue ->
+                                onQrCodeDetected(barcodeValue.replace("radix:", ""))
+                            }
```

Comment:
One thing I would like us to improve later is to make barcode analyser return a flow of elements with distinctUntilChanged configured. This analyser emits values all the time, as long as the camera points to a QR code. We don't need so many emissions since dependant code is executed on each callback.

Reply:
All we can simply add a delay between readings and let the caller decide if distinctUntilChanged is needed.

Reply:
I'll merge it as is now, but let me know if this is something that we should do now (in another PR), IMHO adding the delay is very simple and won't affect the current behavior, but I'm afraid `distinctUntilChanged` approach requires more changes and testing.